### PR TITLE
Add DeleteWorkflowRunLogs function

### LINKED ISSUE #1500: Add support for deleting workflow run logs
Blog post: https://github.blog/changelog/2020-04-21-github-actions-logs-can-now-be-deleted/
API docs: https://developer.github.com/v3/actions/workflow_runs/#delete-workflow-run-logs

I'll open a PR for this soon.

### PR DESCRIPTION
**Pull request by nightlark**
Adds support for the new endpoint added to delete logs.

Doc link: https://developer.github.com/v3/actions/workflow-runs/#delete-workflow-run-logs

Fixes #1500.

### CODE REVIEW

File: github/actions_workflow_runs.go [hunk 1]
```diff
@@ -179,3 +179,17 @@ func (s *ActionsService) GetWorkflowRunLogs(ctx context.Context, owner, repo str
 	parsedURL, err := url.Parse(resp.Header.Get("Location"))
 	return parsedURL, newResponse(resp), err
 }
+
+// DeleteWorkflowRunLogs deletes all logs for a workflow run.
+//
+// GitHub API docs: https://developer.github.com/v3/actions/workflow_runs/#delete-workflow-run-logs
```
Comment (nightlark) [inline github/actions_workflow_runs.go:185]: If the URL change for the GitHub Actions API doc pages was intentional, this needs updating.

```suggestion
// GitHub API docs: https://developer.github.com/v3/actions/workflow-runs/#delete-workflow-run-logs
```

File: github/actions_workflow_runs.go [hunk 2]
```diff
@@ -179,3 +179,17 @@ func (s *ActionsService) GetWorkflowRunLogs(ctx context.Context, owner, repo str
...
 	return parsedURL, newResponse(resp), err
 }
+
+// DeleteWorkflowRunLogs deletes all logs for a workflow run.
+//
+// GitHub API docs: https://developer.github.com/v3/actions/workflow-runs/#delete-workflow-run-logs
+func (s *ActionsService) DeleteWorkflowRunLogs(ctx context.Context, owner, repo string, runID int64) (*Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/actions/runs/%v/logs", owner, repo, runID)
+
+	req, err := s.client.NewRequest("DELETE", u, nil)
```
Comment (oleg-balunenko) [inline github/actions_workflow_runs.go:189]: use constant from net/htt package for method name
https://golang.org/pkg/net/http/#pkg-constants

File: github/actions_workflow_runs_test.go [hunk 3]
```diff
@@ -235,3 +235,18 @@ func TestActionService_ListRepositoryWorkflowRuns(t *testing.T) {
 	}
 
 }
+
+func TestActionService_DeleteWorkflowRunLogs(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/actions/runs/399444496/logs", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "DELETE")
```
Comment (oleg-balunenko) [inline github/actions_workflow_runs_test.go:244]: the same about constant for method name

File: github/actions_workflow_runs.go (hunk 2 above)
Comment (nightlark) [reply github/actions_workflow_runs.go:189]: @gmlewis do we want to switch to using the constants in `net/http` throughout go-github?

Comment (gmlewis) [reply github/actions_workflow_runs.go:189]: I have mixed feelings.  I know that there are a few cases in this repo where we do use the `net/http` constants, but there are very few.

I'm thinking we should proceed with this PR as-is for consistency with the rest of majority of the repo, and then maybe we could accept a separate PR from @oleg-balunenko or anyone else who wants to go through and consistently use the constants throughout the repo.

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Pull request merged**